Description: Fix GCC detection in PGO script
Author: Andrea Pappacoda <andrea@pappacoda.it>
Forwarded: https://codeberg.org/dnkl/foot/pulls/1202
Last-Update: 2022-10-23

diff --git a/pgo/pgo.sh b/pgo/pgo.sh
index b2ce7fe..51a6922 100755
--- a/pgo/pgo.sh
+++ b/pgo/pgo.sh
@@ -30,7 +30,7 @@ do_pgo=no
 CFLAGS="${CFLAGS-} -O3"
 
 case $(${CC-cc} --version) in
-    *GCC*)
+    *Free\ Software\ Foundation*)
         compiler=gcc
         do_pgo=yes
         ;;
